SlackBotPrototype/API/DarkSkyWeatherProvider.cs

```diff
@@ -0,0 +1,44 @@
+﻿using System;
+using System.Linq;
+using Newtonsoft.Json;
+
+namespace API
+{
+
+	public class DarkSkyWeatherProvider : IWeatherProvider
+	{
+		private readonly string _apiSecret;
+		private readonly IHttpRequestClient _client;
+
+		public DarkSkyWeatherProvider(string apiSecret, IHttpRequestClient client)
+		{
+			_apiSecret = apiSecret;
+			_client = client;
+		}
+
+		public string GetForecast(Location location, DateTime when)
+		{
+			var forecastItem = GetForecastRaw(location, when);
+			return forecastItem.Summary;
+		}
+
+		public ChatBotForecast GetForecastRaw(Location location, DateTime when)
+		{
+			
+			var timeStamp = when.DateTimeToUnixTimestamp();
+			// SHORTCUT: We could cache requests in memory for recent forecast requests so we don't have

```

Comment:
Agreed, caching makes sense here especially considering the Dark Sky API is metered. If we're using a cloud service that charges per request we can see the cost savings easily.

Since IO is notoriously a performance bottleneck, anything that does IO is a prime candidate for caching (network accessed services, file access, DB access). At the end of the day, caching is an optimization that adds complexity (invalidation logic) so often it's better to wait until we identify performance issues before making the trade off.